### DISCUSSION TITLE
Allow for selection of schema dynamically at query time

### DISCUSSION THREAD

**Discussion by robot-head** [SeaQL/sea-orm#2117] [Ideas]
We have a multi-tenant application we are working on, and we'd like to select the postgres schema to execute our queries against at runtime.

It seems that the `schema_name` can be configured as part of an `Entity` trait implementation, but we were hoping this could be supplied optionally to the query methods: https://docs.rs/sea-orm/latest/sea_orm/entity/trait.EntityName.html#method.schema_name

**Comment by tyt2y3** [comment]
I think the easiest way is actually leaving the `fn schema` method `None`, but specifying the schema in the connection string with `currentSchema`?

**Reply by robot-head** [reply]
The problem with that is we would need to maintain a connection pool per tenant, and we'd like to support upwards of 50 tenants.  We will be using the same user for all schemas, we just want to switch the schema at runtime without instantiating a new connection pool
